src/Application.vala

```diff
@@ -217,17 +217,18 @@ public class Terminal.Application : Gtk.Application {
         unowned var options = command_line.get_options_dict ();
         var window = (MainWindow) active_window;
         bool new_window;
+        if (window == null ||
+            options.lookup ("new-window", "b", out new_window) && new_window) {
 
-        if (window == null || options.lookup ("new-window", "b", out new_window) && new_window) {
-            /* Uncertain whether tabs should be restored when app is launched with working directory from commandline.
-             * Currently they are set to restore (subject to the restore-tabs setting).
-             * If it is desired that tabs should never be restored in these circimstances add another check below.
-             */
-            bool restore_tabs = !("commandline" in options || "execute" in options) || window == null;
-            window = new MainWindow (this, restore_tabs);
+            window = new MainWindow (this);
         }
 
-        unowned var working_directory = command_line.get_cwd ();
+        unowned string working_directory;
+        var wd_was_requested = options.lookup ("working-directory", "^&ay", out working_directory);
+        working_directory = command_line.get_cwd ();
+
+        assert (working_directory != "\0");  // Can we be sure of this? If not, need fallback
+
```

Comment:
as explained, working-directory isn't meant to be a real option, also assert here isn't right, as it will force quit the primary instance.

Reply:
I'll remove the assert before merge - that is for testing. So far I havent been able to trigger it so probably do not need a fallback?